src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -4543,53 +4545,94 @@ private StatementAst EnumDefinitionRule(List<AttributeBaseAst> customAttributes,
                 return new ErrorStatementAst(enumToken.Extent);
             }
 
-            SkipNewlines();
-            Token lCurly = NextToken();
-            if (lCurly.Kind != TokenKind.LCurly)
+            TypeConstraintAst underlyingTypeConstraint = null;
+            var oldTokenizerMode = _tokenizer.Mode;
+            try
             {
-                // ErrorRecovery: If there is no opening curly, assume it hasn't been entered yet and don't consume anything.
+                SetTokenizerMode(TokenizerMode.Signature);
+                Token colonToken = PeekToken();
+                if (colonToken.Kind == TokenKind.Colon)
+                {
+                    this.SkipToken();
+                    SkipNewlines();
+                    ITypeName underlyingType;
+                    Token unused;
+                    underlyingType = this.TypeNameRule(allowAssemblyQualifiedNames: false, firstTypeNameToken: out unused);
+                    if (underlyingType == null)
+                    {
+                        ReportIncompleteInput(After(colonToken),
+                            nameof(ParserStrings.TypeNameExpected),
+                            ParserStrings.TypeNameExpected);
+                    }
+                    else
+                    {
+                        var resolvedType = underlyingType.GetReflectionType();
+                        if (resolvedType == null || !validUnderlyingTypeCodes.HasFlag(resolvedType.GetTypeCode()))
+                        {
+                            ReportError(underlyingType.Extent,
+                                nameof(ParserStrings.InvalidUnderlyingType),
```

Comment:
Please align arguments in the new code.
(Please push your removed style commit in new PR)

Reply:
Done. Will submit new PR once this is done (to avoid merge conflicts with modified code paths in `Parser.cs`)